### PR TITLE
home work 7.1

### PR DESCRIPTION
**Pull request by graveman13**

### CODE REVIEW

File: src/main/java/core/basesyntax/ArrayList.java [hunk 1]
```diff
@@ -1,53 +1,106 @@
...
-
+        refrashList();
+        list[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isIndexAcceptableLimits(index);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: suggested name: checkIndex

File: src/main/java/core/basesyntax/ArrayList.java [hunk 2]
```diff
@@ -1,53 +1,106 @@
...
+        refrashList();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: suggestion: resizeIfFull

File: src/main/java/core/basesyntax/ArrayList.java [hunk 3]
```diff
@@ -1,53 +1,106 @@
...
+        for (int i = size - 1; i >= size - (size - index); i--) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: try to use System.arraycopy

File: src/main/java/core/basesyntax/ArrayList.java [hunk 4]
```diff
@@ -1,53 +1,106 @@
...
+        isIndexAcceptableLimits(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        isIndexAcceptableLimits(index);
+        T item = (T) list[index];
+        for (int i = index; i < size; i++) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: try to use System.arraycopy

File: src/main/java/core/basesyntax/ArrayList.java [hunk 5]
```diff
@@ -1,53 +1,108 @@
...
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index);
+        resizeIfFull();
+        Object[] tmp = new Object[size + 1];
+        System.arraycopy(list, 0, tmp, 0, size - (size - 1));
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: better move resize logic to a separate method

File: src/main/java/core/basesyntax/ArrayList.java [hunk 6]
```diff
@@ -1,53 +1,108 @@
...
+        checkIndex(index);
+        list[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Object[] tmp = new Object[size - index - 1];
+        System.arraycopy(list, index + 1, tmp, 0, size - index - 1);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/ArrayList.java]: try to use only one arraycopy method call

File: src/main/java/core/basesyntax/ArrayList.java [hunk 7]
```diff
@@ -1,53 +1,109 @@
...
+    private void resizeIfFull() {
+        if (!isCapacityLengthEnsure()) {
+            list = Arrays.copyOf(list, (list.length * 3) / 2 + 1);
+        }
+    }
+
+    private void resizeOnOneElement(int index, int marker) {
+        checkIndex(index);
+        resizeIfFull();
+        System.arraycopy(list, index + (marker < 0 ? 1 : 0), list,
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/ArrayList.java]: can we simplify this logic?

**Pull request closed without merging**